src/js/utils/index.d.ts

```diff
@@ -35,16 +35,17 @@ declare const removeUndefined: <T extends object>(obj: T) => NonUndefinedProps<T
 export {isObject, deepFreeze, deepMerge, removeUndefined};
 
 // Extracting types for common properties among components
+type BoxSideType = "top" | "left" | "bottom" | "right" | "horizontal" | "vertical" | "all";
+type BoxSizeType = "xsmall" | "small" | "medium" | "large" | "xlarge" | string;
+type BoxStyleType = "solid" | "dashed" | "dotted" | "double" | "groove" | "ridge" | "inset" | "outset" | "hidden";
```

Comment:
I understand why this got moved up as they are only being used internally and not exported like the other types below, but I’m not sure I fully understand why they got renamed and why the naming chosen was `Box` instead of `Border`. Not a big deal, obviously, I’m just seeking understanding from people who have the bigger picture in view.

Reply:
Good question. 
It was important to me to make it clear that the types you see there of BoxSideType, BoxSizeType, BoxStyleType should only be associated with Box component. 
For example, some other components other than Box might have `size` but it will use different values and not the exact Box sizing, so I added the extra description to type so no one will be tempted to use those props for components that are not underlined with the Box component.
I agree this might need further clean-ups, as the code grows we grow to understand the delicate handling of typescript as well.

BTW, not sure if you saw the slack message, but grommet 2.8.1 is out and this PR was a great reason to trigger it 🎉